Assets/Scripts/Models/Character.cs

```diff
@@ -1017,9 +999,9 @@ private bool CheckForJobMaterials()
 
     private bool WalkingToUsableInventory()
     {
-        bool destHasInventory = pathAStar != null && pathAStar.EndTile() != null && pathAStar.EndTile().Inventory != null;
+        bool destHasInventory = movementPath != null && movementPath.Last() != null && movementPath.Last().Inventory != null;
```

Comment:
Seems like the real problem is in the job handling code but I'm not sure what's really happening without a good way to reproduce the problem.